actors/evm/src/interpreter/instructions/call.rs

```diff
@@ -157,50 +164,77 @@ pub fn call<BS: Blockstore, RT: Runtime<BS>>(
             let dst_addr: EthAddress = dst.try_into()?;
             let dst_addr: Address = dst_addr.try_into()?;
 
-            let call_result = match kind {
-                CallKind::Call => system.send(
-                    &dst_addr,
-                    // readonly is sticky
-                    if system.readonly {
-                        Method::InvokeContractReadOnly
-                    } else {
-                        Method::InvokeContract
-                    } as u64,
-                    // TODO: support IPLD codecs #758
-                    RawBytes::serialize(BytesSer(input_data))?,
-                    TokenAmount::from(&value),
-                ),
-
-                CallKind::DelegateCall => {
-                    // first invoke GetBytecode to get the code CID from the target
-                    let code = crate::interpreter::instructions::ext::get_evm_bytecode_cid(
-                        system.rt, dst,
-                    )?;
-
-                    // and then invoke self with delegate; readonly context is sticky
-                    let params = DelegateCallParams {
-                        code,
-                        input: input_data.to_vec(),
-                        readonly: system.readonly,
-                    };
-                    system.send(
-                        &system.rt.message().receiver(),
-                        Method::InvokeContractDelegate as u64,
-                        RawBytes::serialize(&params)?,
-                        TokenAmount::from(&value),
-                    )
+            // Special casing for embryo/non-existent actors: we just do a SEND (method 0)
+            // which allows us to transfer funds (and create embryos)
+            let is_embryonic = if dst_addr.protocol() == AddressProtocol::ID {
+                // sanity check: this shouldn't be an ID address, as you can't predict
+                // what actor is gonna sit there.
+                false
```

Comment:
You can. Once an embryo exists, it has an ID address.

Basically, we need to:

1. Resolve the address.
2. Lookup the code CID for the address.
3. If it doesn't exist or is an embryo, send with method 0.
4. Otherwise, method 2.

Reply:
@vyzo https://github.com/filecoin-project/builtin-actors/pull/775